misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java

```diff
@@ -920,6 +956,645 @@ public NilTypeDescriptorNode transform(NilTypeDescriptorNode nilTypeDescriptorNo
+                this.trailingWS, this.trailingNL);
+
+        return functionTypeDescriptorNode.modify()
+                .withQualifierList(qualifierList)
+                .withFunctionKeyword(functionKeyword)
+                .withFunctionSignature(functionSignature)
+                .apply();
+    }
+
+    @Override
+    public ParenthesisedTypeDescriptorNode transform(ParenthesisedTypeDescriptorNode parenthesisedTypeDescriptorNode) {
+        Token openParenToken = formatToken(parenthesisedTypeDescriptorNode.openParenToken(), 0, 0);
+        TypeDescriptorNode typedesc = formatNode(parenthesisedTypeDescriptorNode.typedesc(), 0, 0);
+        Token closeParenToken = formatToken(parenthesisedTypeDescriptorNode.closeParenToken(),
+                this.trailingWS, this.trailingNL);
+
+        return parenthesisedTypeDescriptorNode.modify()
+                .withOpenParenToken(openParenToken)
+                .withTypedesc(typedesc)
+                .withCloseParenToken(closeParenToken)
+                .apply();
+    }
+
+    @Override
+    public ExternalFunctionBodyNode transform(ExternalFunctionBodyNode externalFunctionBodyNode) {
+        Token equalsToken = formatToken(externalFunctionBodyNode.equalsToken(), 1, 0);
+        NodeList<AnnotationNode> annotations = formatNodeList(externalFunctionBodyNode.annotations(), 0, 1, 1, 0);
+        Token externalKeyword = formatToken(externalFunctionBodyNode.externalKeyword(), 0, 0);
+        Token semicolonToken = formatToken(externalFunctionBodyNode.semicolonToken(), this.trailingWS, this.trailingNL);
+
+        return externalFunctionBodyNode.modify()
+                .withEqualsToken(equalsToken)
+                .withAnnotations(annotations)
+                .withExternalKeyword(externalKeyword)
+                .withSemicolonToken(semicolonToken)
+                .apply();
+    }
+
+    @Override
+    public AnnotationNode transform(AnnotationNode annotationNode) {
+        Token atToken = formatToken(annotationNode.atToken(), 0, 0);
+        Node annotReference = formatNode(annotationNode.annotReference(), 1, 0);
+
+        if (annotationNode.annotValue().isPresent()) {
+            MappingConstructorExpressionNode annotValue = formatNode(annotationNode.annotValue().get(),
+                    this.trailingWS, this.trailingNL);
+            annotationNode = annotationNode.modify().withAnnotValue(annotValue).apply();
+        }
+
+        return annotationNode.modify()
+                .withAtToken(atToken)
+                .withAnnotReference(annotReference)
+                .apply();
+    }
+
+    @Override
+    public MappingConstructorExpressionNode transform(
+            MappingConstructorExpressionNode mappingConstructorExpressionNode) {
+        Token openBrace = formatToken(mappingConstructorExpressionNode.openBrace(), 0, 1);
+        indent();
+        SeparatedNodeList<MappingFieldNode> fields = formatSeparatedNodeList(
```

Comment:
We could try to do something similar to the record-type-desc, to selectively expand